### DISCUSSION TITLE
Pyright Inference Algorithm

### DISCUSSION THREAD

**Discussion by ahmedihabb2** [microsoft/pyright#4825] [General]
Hello All,
I want to know what algorithm does Pyright used for inference, and if there is any paper or something that explains it
As I want to understand it or probably implement a type inference algorithm for learning purposes.

**Comment by erictraut** [comment]
Pyright's inference behavior is documented [here](https://microsoft.github.io/pyright/#/type-concepts-advanced). If you want more details, pyright's code is open and is published [here](https://github.com/Microsoft/pyright).
